holidays/countries/japan.py

```diff
@@ -51,29 +51,34 @@ def __init__(self, *args, **kwargs) -> None:
         kwargs.setdefault("observed_rule", SUN_TO_NEXT_WORKDAY)
         super().__init__(*args, **kwargs)
 
+    def _is_observed(self, dt: date) -> bool:
+        return dt >= date(1973, APR, 12)
+
     def _populate_observed(self, dts: Set[date]) -> None:
         # When a national holiday falls on Sunday, next working day
-        # shall become a public holiday (振替休日) - substitute holidays.
+        # shall become a public holiday (振替休日) - substitute holiday.
         for dt in sorted(dts):
-            is_observed, dt_observed = self._add_observed(
-                dt,
-                # Substitute Holiday.
-                name=tr("振替休日"),
-                show_observed_label=False,
-            )
-            if is_observed:
-                dts.add(dt_observed)  # type: ignore[arg-type]
-
-        # A weekday between national holidays becomes
-        # a holiday too (国民の休日) - national holidays.
-        for dt in dts:
-            if _timedelta(dt, +2) not in dts:
-                continue
-            dt_observed = _timedelta(dt, +1)
-            if self._is_sunday(dt_observed) or dt_observed in dts:
-                continue
-            # National Holiday.
-            self._add_holiday(tr("国民の休日"), dt_observed)
+            # Substitute Holiday.
+            self._add_observed(dt, name=tr("振替休日"), show_observed_label=False)
+
+        # A weekday between national holidays becomes a holiday too (国民の休日) -
+        # national holiday.
+        # In 1986-2006 it was only May 4 (between Constitution Day and Children's Day).
+        # Since 2006, it may be only the day between Respect for the Aged Day and
+        # Autumnal Equinox Day (in September).
+        if self._year <= 1985:
+            return None
+        if self._year <= 2006:
```

Comment:
```suggestion
        elif 1986 <= self._year <= 2006:
```

Reply:
But what will we gain from it? 👀

Reply:
Nvm, I thought it was supposed to be part of the same if-else as this section above:
```python
        if self._year <= 1985:
            return None
        if self._year <= 2006:
            may_4 = (MAY, 4)
            if not (self._is_monday(may_4) or self._is_sunday(may_4)):
                # National Holiday.
                self._add_holiday(tr("国民の休日"), may_4)
        else:
            for dt in dts:
                if dt.month == SEP and _timedelta(dt, +2) in dts:
                    # National Holiday.
                    self._add_holiday(tr("国民の休日"), _timedelta(dt, +1))
                    break
```